<a href="https://colab.research.google.com/github/OlegV12/Stats/blob/Lesson_5/mathStats.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from math import factorial, pow, e, pi

In [6]:
import numpy as np
from scipy.stats import norm

In [ ]:

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [ ]:
def bernulli(n, k, p):
  """
  :param n number of observations
  :param k number of occurrences
  :param p probability of occurrence
  """
  return combinations(n, k) * pow(p, k) * pow((1-p), (n-k))

In [ ]:
def poisson(n, k, p):
  """
  :param n number of observations
  :param k number of occurrences
  :param p probability of occurrence
  """
  a = n * p # lambda

  return pow(a, k) / factorial(k) * pow(e, (-a))



In [ ]:
def mean_sqrt_dev(a):
  res = []
  mean = sum(a) / len(a)
  for i in a:
    res.append(pow((i - mean), 2))
  return np.sqrt(sum(res) / len(a)), np.sqrt(sum(res) / (len(a)-1))


In [ ]:
def prob(mu, x, sqd):
  return (x-mu)/sqd

Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [14]:
a1 = 80 - (1.96 * (16 / np.sqrt(256)))
a1

78.04

In [18]:
a2 = 80 + (1.96 * (16 / np.sqrt(256)))
a2

81.96

Ответ: [78.04, 81.96]

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [27]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
x_mean = 6.59


In [38]:
a1 = x_mean - (2.262 * (np.std(x, ddof=1)/np.sqrt(10)))
a1

6.2675382559124255

In [39]:
a2 = x_mean + (2.262 * (np.std(x, ddof=1)/np.sqrt(10)))
a2

6.912461744087574

Ответ: с вероятностью 95% истинное значение величины Х лежит в интервале [6.267, 6.912]

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

H0: mu == mu0

H1: mu > mu0

In [41]:
z_t = 1.65

In [43]:
z = (17.5 - 17 ) / (np.sqrt(4) / np.sqrt(100))  
z

2.5

In [44]:
z > z_t

True

Ответ: верна альтернативная гипотеза

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

H0: mu == mu0

H1: mu != mu0

In [56]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [57]:
t_t = 3.249

In [58]:
x_std = np.std(x, ddof=1)
t = (np.mean(x) - 200 ) / (x_std / np.sqrt(10))
t

-1.0651074037450896

Ответ: т.к. t-расчетное лежит в интервале от -3.249 сигм до +3.249 сигм, принимается нулевая гипотеза, утверждение продавца верно. 
